Kshitij/py3

xitij2000 opened the pull request.

giovannicimolin commented on tests/integration/test_chat.py:
```diff
@@ -1,14 +1,12 @@
 import os
```
Running `make isort` on Python 2.7 here yields:
```
diff --git a/tests/integration/test_chat.py b/tests/integration/test_chat.py
index 75fe128..d3b7d06 100644
--- a/tests/integration/test_chat.py
+++ b/tests/integration/test_chat.py
@@ -3,14 +3,15 @@ import re
 
 import pkg_resources
 import yaml
-from bok_choy.promise import EmptyPromise
 from ddt import data, ddt
 from django.test.client import Client
-from mock import ANY, patch
 from xblock.reference.user_service import XBlockUser
 from xblockutils.resources import ResourceLoader
 from xblockutils.studio_editable_test import StudioEditableBaseTest
 
+from bok_choy.promise import EmptyPromise
+from mock import ANY, patch
+
 loader = ResourceLoader(__name__)
```

xitij2000 replied:
The above seems incorrect. I don't see why those lines should move. 

Given that one solutions is upgraded to Juniper there is no need for a python2 version, we can stick to the Python3 path.

giovannicimolin reviewed (changes requested):
@xitij2000 I've tested this in the `master` devstack, and it seems to be working well.

When checking the CI, I've seem issues like these on all three test runs:
```
django.db.utils.OperationalError: database table is locked: workbench_xblockstate
WARNING:django.request:Bad Request: /handler/test-scenario.chat.d3.u0/submit_studio_edits/
```
Not sure what's causing them, but the previous tests didn't had this issue.

giovannicimolin reviewed (changes requested):
@xitij2000 Though the tests appear to pass, there's something wrong with the CI run. 
I see these issues:
**1.** 
```
ERROR:django.request:Internal Server Error: /handler/test-scenario.chat.d0.u0/chat_complete/
Traceback (most recent call last):
  File "/home/circleci/project/venv/lib/python3.5/site-packages/django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "/home/circleci/project/venv/lib/python3.5/site-packages/django/core/handlers/base.py", line 115, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/home/circleci/project/venv/lib/python3.5/site-packages/django/core/handlers/base.py", line 113, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/circleci/project/venv/src/xblock-sdk/workbench/views.py", line 123, in handler
    return webob_to_django_response(result)
  File "/home/circleci/project/venv/lib/python3.5/site-packages/xblock/django/request.py", line 18, in webob_to_django_response
    webob_response.app_iter,
AttributeError: 'dict' object has no attribute 'app_iter'
ERROR:django.request:Internal Server Error: /handler/test-scenario.chat.d0.u0/chat_complete/
Traceback (most recent call last):
  File "/home/circleci/project/venv/lib/python3.5/site-packages/django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "/home/circleci/project/venv/lib/python3.5/site-packages/django/core/handlers/base.py", line 115, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/home/circleci/project/venv/lib/python3.5/site-packages/django/core/handlers/base.py", line 113, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/circleci/project/venv/src/xblock-sdk/workbench/views.py", line 123, in handler
    return webob_to_django_response(result)
  File "/home/circleci/project/venv/lib/python3.5/site-packages/xblock/django/request.py", line 18, in webob_to_django_response
    webob_response.app_iter,
AttributeError: 'dict' object has no attribute 'app_iter'
```

**2.**
```
WARNING:bok_choy.browser:Could not save browser log of type 'client'. It may be that the browser does not support it.
Traceback (most recent call last):
  File "/home/circleci/project/venv/lib/python2.7/site-packages/bok_choy/browser.py", line 143, in save_driver_logs
    log = driver.get_log(log_type)
  File "/home/circleci/project/venv/lib/python2.7/site-packages/selenium/webdriver/remote/webdriver.py", line 1053, in get_log
    return self.execute(Command.GET_LOG, {'type': log_type})['value']
  File "/home/circleci/project/venv/lib/python2.7/site-packages/selenium/webdriver/remote/webdriver.py", line 252, in execute
    self.error_handler.check_response(response)
  File "/home/circleci/project/venv/lib/python2.7/site-packages/selenium/webdriver/remote/errorhandler.py", line 165, in check_response
    raise exception_class(value)
WebDriverException: Message: HTTP method not allowed
```

I'm not too worried about _2_ tough. Is the first one something we're deliberately testing?

xitij2000 commented:
@giovannicimolin I've upgraded firefox to the latest ESR and fixed a bunch of other stuff. I've also moved out the quality checks to a separate job to improve parallelism and segregate quality checks from tests, to avoid duplicate test runs and get true parallel tests.

I've switched to using pytest, and since it was low effort I also integrated with CircleCi's test reporting system so you can see passed and failed tests in theirr UI.

giovannicimolin reviewed (approved):
:+1:

- [x] I tested this: 
1. Tested the block in the master devstack.
2. Checked that all test runs are green, and run on environments for each platform version (ironwood, juniper, koa)
- [x] I read through the code
- [ ] ~~I checked for accessibility issues~~ NA
- [ ] ~~Includes documentation~~ NA

@xitij2000 Good to go!
